### PR TITLE
HPCC-12882 Fix various warnings and problems with the regression suite

### PR DESCRIPTION
**Pull request by ghalliday**
Changes include
- Merging two error level enumerations
- Merging jerrorrange.h and errorlist.h
- Adding warning numbers to various eclagent/hthor warnings
- Use the warning mappings in eclagent to allow warnings to be removed.
- Add #onwwarning to remove compile and execution warnings
- Fixed/split a few versions of tests that couldn't work with index translation

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review

File: common/fileview2/fverror.hpp [hunk 1]
```diff
@@ -18,41 +18,43 @@
 #ifndef FVERROR_HPP
 #define FVERROR_HPP
 
-#include "jerrorrange.hpp"
+#include "errorlist.h"
 
-#define ERR_FILEVIEW_FIRST  2000
-#define ERR_FILEVIEW_LAST   2049
-
-#define FVERR_CouldNotResolveX                  2000
```
Comment (richardkchapman) [inline common/fileview2/fverror.hpp]: I don't like this change. If I need to look up what error code 2002 means, I won't find it

File: common/fileview2/fverror.hpp [hunk 2]
```diff
@@ -18,41 +18,43 @@
...
-#define FVERR_ExpectedFieldSelectedFromRecordXX 2021
-#define FVERR_NumJoinFieldsMismatchXY           2022
-#define FVERR_ExpectedX                         2023
-#define FVERR_FailTransformation                2024
-#define FVERR_UnrecognisedMappingFunctionXY     2025
-#define FVERR_BadStringTermination              2026
-#define FVERR_CannotBrowseFile                  2027
-#define FVERR_PluginMismatch                    2028
-#define FVERR_RowTooLarge                       2029
+//Don't remove items from this list - otherwise the error codes will change
```
Comment (richardkchapman) [inline common/fileview2/fverror.hpp]: Another reason I don't like this change

**Comment by ghalliday** [comment]
Note: Also requires https://github.com/hpcc-systems/LN/pull/722 dues to rename of enumerations

File: common/fileview2/fverror.hpp (hunk 2 above)
Comment (ghalliday) [inline common/fileview2/fverror.hpp]: Fair enough - I'll change those back

**Comment by ghalliday** [comment]
@richardkchapman switched back to using #defines.  Some of the ranges changed because the two files clashed or ranges weren't being used properly.

**Pull request merged**